src/mem/ruby/system/CacheRecorder.cc

```diff
@@ -54,14 +57,29 @@ CacheRecorder::CacheRecorder()
 {
 }
 
+#if BUILD_GPU
```

Comment:
shouldn't this be #ifdef instead of #if?

Reply:
At the start of compile time, configs/build_gpu.hh is generated with a value of either 0 or 1 (if VEGA_X86) for  BUILD_GPU. #ifdef returns true always since BUILD_GPU is defined in either case. I used #if to differentiate between its possible values 0 and 1.

Reply:
I am not sure I understand the distinction, but ok.

Reply:
I'm unconvinced this was a safe thing to do. It is bad practice to rely on magic values like this! 

In fact, I'm pretty sure that #69 will break this change.

I'm quite concerned about adding more build-time guard flags. We are trying to move away from configuring gem5 at build time (see change for multi-ISA build (already merged) and multi-Ruby build (not yet merged)).

Reply:
Do you have a recommendation about an alternative then?

Reply:
I am not sure I understand the implicit/explicit part, because BUILD_GPU is explicitly set to 1 in the case you described.  I assume you mean it is not explicitly set to 0 in other builds?

Ultimately from @abmerop above, this is the problem:

"I think it is due to the ALL build not building the GPU, and the DebugFlags for GPUMem, etc. are guarded in src/gpu-compute/SConscript are guarded by the configuration variable BUILD_GPU.

I assume this means that all GPU patches will now fail tests if that's the case."

The only options I can think of are:

1. Don't guard the GPU DebugFlags with BUILD_GPU -- I guess there is no harm in them being accessible in a build that is not for the GPU?
2. Have ALL build the GPU -- not sure if there is a Board/stdlib related reason for this though?
3. Use something like ifdef's and explicitly set BUILD_GPU to false in things that aren't GCN3_X86/VEGA_X86.